sdk/python/kfp/dsl/__init__.py

```diff
@@ -13,9 +13,9 @@
 # limitations under the License.
 
 
-from ._pipeline_param import PipelineParam
+from ._pipeline_param import PipelineParam, _match_serialized_pipelineparam
```

Comment:
Why is it needed to be exposed? If it's really needed, can it be named as a public method?

Reply:
I need it in `compiler` to extract and replace the serialized pipelineparam with `input.parameters.%s`. Did not want to have 2 sets of duplicated regex in 2 diff modules.

**Proposed solution:**  
We can probably a `StringWithPipelineParams` class (which holds a string with 1 or more serialized PipelineParams) instead of having a mix of strings and `PipelineParam`.

And add `to_argo_variable` method to both classes.